boards/weio/system.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     boards_weio
+ * @{
+ *
+ * @file
+ * @brief       Board specific implementations for the WeIO board
+ *
+ * @author      Paul RATHGEB <paul.rathgeb@skynet.be>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+#include "LPC11Uxx.h"
+
+#define SYSOSCCTRL_Val    0x00000000 // Reset: 0x000
+#define SYSPLLCTRL_Val    0x00000023 // Reset: 0x000
+#define SYSPLLCLKSEL_Val  0x00000001 // Reset: 0x000
+#define MAINCLKSEL_Val    0x00000003 // Reset: 0x000
+#define SYSAHBCLKDIV_Val  0x00000001 // Reset: 0x001
+#define USBPLLCTRL_Val    0x00000023 // Reset: 0x000
+#define USBPLLCLKSEL_Val  0x00000001 // Reset: 0x000
+#define USBCLKSEL_Val     0x00000000 // Reset: 0x000
+#define USBCLKDIV_Val     0x00000001 // Reset: 0x001
+
+/* Define clocks */
+#define __XTAL            (12000000UL)    /* Oscillator frequency */
+#define __SYS_OSC_CLK     (    __XTAL)    /* Main oscillator frequency */
+
+#define __SYS_PLLCLKIN    (__SYS_OSC_CLK)
+#define  __SYS_PLLCLKOUT  (__SYS_PLLCLKIN * ((SYSPLLCTRL_Val & 0x01F) + 1))
+#define __MAIN_CLOCK      (__SYS_PLLCLKOUT)
+#define __SYSTEM_CLOCK    (__MAIN_CLOCK / SYSAHBCLKDIV_Val)
+
+uint32_t SystemCoreClock = __SYSTEM_CLOCK; /*!< System Clock Frequency (Core Clock)*/
+
+/**
+ * Initialize the system
+ *
+ * @param  none
+ * @return none
+ *
+ * @brief  Setup the microcontroller system.
+ *         Initialize the System.
+ */
+void SystemInit (void) {
```

Comment:
Ok, I would still vote to take this into cpu for now. We can then think about a way to separate this once there is a second board using this MCU.